api/src/watersystem/get.handler.ts

```diff
@@ -0,0 +1,89 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+import * as AWS from 'aws-sdk';
+import { RDSDataService } from 'aws-sdk';
+import {
+  ExecuteStatementRequest,
+  FieldList,
+  SqlParametersList,
+} from 'aws-sdk/clients/rdsdataservice';
+
+const CORS_HEADERS = {
+  'Access-Control-Allow-Headers': 'Content-Type',
+  'Access-Control-Allow-Origin': '*',
+  'Access-Control-Allow-Methods': 'OPTIONS,GET',
+};
+const SCHEMA = 'public';
+
+async function getWaterSystemData(
+  rdsService: RDSDataService,
+  params: SqlParametersList,
+): Promise<WaterSystemApiResponse> {
+  const executeParams: ExecuteStatementRequest = {
+    database: process.env.DATABASE_NAME ?? 'postgres',
+    resourceArn: process.env.RESOURCE_ARN ?? '',
+    schema: SCHEMA,
+    secretArn: process.env.CREDENTIALS_SECRET ?? '',
+    sql: `SELECT pws_id,
+                 lead_connections_count,
+                 service_connections_count
+          FROM water_systems
+          WHERE pws_id = :pws_id LIMIT 1`,
+    parameters: params,
+  };
+  let body: WaterSystemApiResponse = {};
+
+  const results = await rdsService.executeStatement(executeParams).promise();
+  for (let record of results.records ?? []) {
+    body = {
+      pws_id: record[0].stringValue,
+      lead_service_lines: record[1].doubleValue,
+      service_lines: record[2].doubleValue,
+    };
+  }
+  return body;
+}
+
+export const handler = async (event: {
+  pathParameters: WaterSystemPathParameters;
+}): Promise<APIGatewayProxyResult> => {
+  const pwsId = event.pathParameters?.pws_id ?? '';
+
+  const db = new AWS.RDSDataService();
+  const params: SqlParametersList = [
+    {
+      name: 'pws_id',
+      value: { stringValue: pwsId },
+    },
+  ];
+  let body: WaterSystemApiResponse;
+
+  try {
+    body = await getWaterSystemData(db, params);
+  } catch (error) {
+    console.log(`Error fetching water system data for pws id ${pwsId}`, error);
```

Comment:
should we return not 200 error in this case? not sure what best practice is

Reply:
This would happen if there's an internal error or an exception through by RDS service, so we would want to return a 500. According to [this](https://aws.amazon.com/blogs/compute/error-handling-patterns-in-amazon-api-gateway-and-aws-lambda/) an error thrown should get passed through the response and then we can map certain responses to statuses. I'm adding a task to the backlog to see if more work needs to be done here to get errors handled elegantly. Thanks for raising!